gdx/src/com/badlogic/gdx/graphics/g3d/loader/ObjLoader.java

```diff
@@ -243,9 +243,10 @@ protected ModelData loadModelData (FileHandle file, boolean flipV) {
 			if (hasNorms) attributes.add(new VertexAttribute(Usage.Normal, 3, ShaderProgram.NORMAL_ATTRIBUTE));
 			if (hasUVs) attributes.add(new VertexAttribute(Usage.TextureCoordinates, 2, ShaderProgram.TEXCOORD_ATTRIBUTE + "0"));
 
-			String nodeId = "node" + (++id);
-			String meshId = "mesh" + id;
-			String partId = "part" + id;
+			String stringId = "" + (++id);
+			String nodeId = group.name != "default" ? group.name : "node" + stringId;
+			String meshId = group.name != "default" ? group.name : "mesh" + stringId;
+			String partId = group.name != "default" ? group.name : "part" + stringId;
```

Comment:
Just out curiosity, did you test this on all platforms and did it actually work? I'd though that at least one of gwt or robovm to fail.

Anyway, can you change this to `equals` please. http://stackoverflow.com/questions/513832/how-do-i-compare-strings-in-java